src/role_utils.rs

```diff
@@ -0,0 +1,127 @@
+//! This module provides utility functions for dealing with role (types) and role groups.
+//!
+//! While other modules in this crate try to be generic and reusable for other operators
+//! this one makes very specific assumptions about how a CRD is structured.
+//!
+//! These assumptions are detailed and explained below.
+//!
+//! # Roles / Role types
+//!
+//! A CRD is often used to operate another piece of software.
+//! Software - especially the distributed kind - sometimes consists of multiple different types of program working together to achieve their goal.
+//! These different types is what we call a _role_.
+//!
+//! ## Examples
+//!
+//! Apache Hadoop HDFS:
+//! * NameNode
+//! * DataNode
+//! * JournalNode
+//!
+//! Kubernetes:
+//! * kube-apiserver
+//! * kubelet
+//! * kube-controller-manager
+//! * ...
+//!
+//! # Role Groups
+//!
+//! There is sometimes a need to have different configuration options or different label selectors for different instances of the same role.
+//! Role groups are what allows this.
+//! Nested under a role there can be multiple role groups, each with its own LabelSelector and configuration.
+//!
+//! ## Example
+//!
+//! Role Groups: `default`, `20-cores`, `gpu`
+//!
+//! # Pod labels
+//!
+//! Each Pod that Operators create needs to have a common set of labels:
+//! * <TODO>/name - The name of the parent resource, this is useful so an operator can list all its pods by using a LabelSelector
```

Comment:
Opinions on how these labels should look like? @soenkeliebau

Reply:
Not really, for the Kafka operator my current idea is:

```
kafka.stackable.tech/service
kafka.stackable.tech/role-group
kafka.stackable.tech/role
```

But nothing here is anything that I am passionate about..